File: SchoolWebApp-API/Project.Core/Entities/School/ToDoList.cs
using SchoolWebApp.Core.Entities.Shared;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SchoolWebApp.Core.Entities.School
{
    public class ToDoList: Base
    {
        public required string ItemName { get; set; }
        public DateTime CompleteBy { get; set; }
        public Boolean Completed { get; set; }

        public int? UserId { get; set; }

        [NotMapped]
        public double TimeToDeadline
        {
            get
            {
                var dateDiff = CompleteBy - DateTime.Now;
                return dateDiff.TotalMinutes;
            }
        }
    }
}

File: SchoolWebApp-Client/src/app/school/models/todolist.ts
import { ResourceModel } from "@/core/models/ResourceModel";

export class TodoList extends ResourceModel<TodoList> {
    public itemName?: string;
    public completeBy?: Date;
    public completed?: boolean;
    public timeToDeadline: number;

    public staffDetailsId?: number;

    constructor(model?: Partial<TodoList>) {
        super(model);
    }
}
